components/jupyter-web-app/backend/kubeflow_jupyter/common/utils.py

```diff
@@ -387,6 +387,61 @@ def set_notebook_memory(notebook, body, defaults):
     container["resources"]["requests"]["memory"] = memory
 
 
+def set_notebook_gpus(notebook, body, defaults):
+    gpus = None
+    gpuDefaults = defaults.get("gpus", {})
+    if gpuDefaults.get("readOnly", False):
```

Comment:
Does Kubernetes not support any form of introspection that would allow us to avoid explicitly configuring this?

Reply:
The `readOnly` field in JWA's config is for the admin to specify that the users should not be able to edit the values of that field in the UI's form. At this point we make sure to use the default value in the config if the admin has set it to be `readOnly`.

It's not clear to me how we could leverage k8s for this one. Was it a miss interpretation on what this small section of code was doing or am I missing something?

Reply:
@kimwnasptd apologies I think I commented on the wrong line. I meant could we not use introspection to some how get the names for the limit values e.g. "nvidia.com"?

Reply:
Oh I see what you mean. We could maybe check the `node.capacity` or `node.allocatable` to determine the extra available resources on the nodes. But we will need to define how to filter what devices we want to show in JWA.

I also think that this will be the next improvement of this feature but I'd like to dig a little bit more into it first to assess our options.